rfcs/proposed/iterators_pass_directly_customization/README.md

```diff
@@ -0,0 +1,229 @@
+specific ADL customization point function for the derived class. This is particularly useful in cases of multiple
+inheritance or ambiguous ADL matches, where the default behavior may not align with the intended design.
+
+Users must be aware of this behavior. A user might incorrectly assume that the absence of a customization point
+specifically for the derived class would cause the derived iterator type to use the default overload of
+`is_passed_directly_in_onedpl_device_policies` (and return `std::false_type`).
+
+### Implementation Details
+
+When using device policies, oneDPL will run compile-time checks on argument iterator types by using `decltype` to
+determine the return type of `is_passed_directly_in_onedpl_device_policies` when called with iterator as argument. If
+`std::true_type{}` is returned, oneDPL will pass the iterator directly to SYCL kernels rather than copying the data into
+`sycl::buffers` and using those buffers to transfer data to SYCL kernels.
+
+The specification and implementation will be prepared once this RFC is accepted as "proposed". We do not intend to offer
+this first as experimental. This RFC will target "Supported" once the specification and implementation are accepted.
+
+### Examples
+
+Below is a simple example of an iterator and ADL customization point definition which is always "passed directly".
+
+```cpp
+namespace user
+{
+
+    struct my_passed_directly_iterator
+    {
+        /* unspecified user definition */
+    };
+
+    std::true_type
+    is_passed_directly_in_onedpl_device_policies(const my_passed_directly_iterator&)
+    {
+        return std::true_type{};
+    }
+} //namespace user
+```
+
+Users can use any compile-time logic based on their iterator type to determine if it can be "passed directly" to define
+the ADL function's return type. Commonly, this will involve some check of base template types and their "passed
+directly" trait status. Below is an example of a type that contains a pair of iterators and should be treated as "passed
+directly" if and only if both base iterators are also "passed directly". Note that the use of the public trait enables
+the use of the base iterator type alone without creating a named instance of the base iterator to pass into the ADL.
+
+```cpp
+namespace user
+{
+    template <typename It1, typename It2>
+    struct iterator_pair
+    {
+        It1 first;
+        It2 second;
+    };
+
+    template <typename It1, typename It2>
+    auto is_passed_directly_in_onedpl_device_policies(const iterator_pair<It1, It2>&)
+    {
+        return std::conjunction<oneapi::dpl::is_passed_directly_to_device<It1>,
+                                oneapi::dpl::is_passed_directly_to_device<It2>>{};
+    }
+} //namespace user
+```
+
+It is also possible to write overloads without a body using an `auto` trailing return type. The following is an
+alternative written in this way for the previous example:
```

Comment:
It does not matter whether you use a trailing return type of not. It will still work. So, the suggestion is
```suggestion
It is also possible to write overloads without a body. The following is an
alternative written in this way for the previous example:
```